nbs_probabilistic/create_dropout_ensemble.py

```diff
@@ -0,0 +1,149 @@
+import fire
+from fire import Fire
+import xarray as xr
+import numpy as np
+from src.data_generator import *
+from src.train import *
+from src.networks import *
+from src.utils import *
+from tensorflow.keras import backend as K
+#ToDO: 
+#load full data instead of batches. output for full size of X. Stii unable to do. How to I run a code on GPU?
+#make it work for all networks. #(Differences: custom_objects, loss function, ...
+#CouldDo: pass optional arguments. like is_normalized, start_date, end_date
+
+
+def get_input(args, exp_id_path, datadir, model_save_dir):
+    #essential arguments
+    args=load_args(exp_id_path)
+    exp_id=args['exp_id']
+    var_dict=args['var_dict']
+    batch_size=args['batch_size']
+    output_vars=args['output_vars']
+    
+    #optional inputs. see load_args() for default values.
+    data_subsample=args['data_subsample']
+    norm_subsample=args['norm_subsample']
+    nt_in=args['nt_in'] #Ques: sometimes mentioned in config file as 'nt'. is that same?
+    #nt_in=args['nt']
+    dt_in=args['dt_in']
+    test_years=args['test_years']
+    lead_time=args['lead_time']
+    
+    #changing paths
+#   model_save_dir='/home/garg/data/WeatherBench/predictions/saved_models'
+#   datadir='/home/garg/data/WeatherBench/5.625deg'
+    
+    #run data generator
+    ds = xr.merge([xr.open_mfdataset(f'{datadir}/{var}/*.nc', combine='by_coords') 
+                   for var in var_dict.keys()])
+    mean = xr.open_dataarray(f'{model_save_dir}/{exp_id}_mean.nc') 
+    std = xr.open_dataarray(f'{model_save_dir}/{exp_id}_std.nc')
+    
+    ds_test= ds.sel(time=slice(test_years[0],test_years[-1]))
+    #Ques: Should shuffle be False? since its testing.
+    dg_test = DataGenerator(ds_test, var_dict, lead_time, batch_size=batch_size, shuffle=False,
+                            load=True,mean=mean, std=std, output_vars=output_vars,
+                            data_subsample=data_subsample, norm_subsample=norm_subsample,
```

Comment:
I just realized that you DO have to take the `data_subsample` from `args`. This is because most models are trained with `nt_in > 1` and in these cases it is important that the data_subsample parameter is not changed otherwise `nt_in` does not represent the same time step.

Reply:
OMG, I just realized that I have been doing this wrong the entire time! The good thing is that this improves my scores (a tiny little bit) actually. Oh dear...

Reply:
I was also looking at it right now too.
I put data_subsample=6 hours and ran dg_test.valid_time.
 the first entry was 2017-01-04T12:00:00 instead of 2017-01-04T00:00:00. (72 hours)
I'll change.

Reply:
To reduce the time it takes to compute you can change test_years to ['2017', '2017']